### PR TITLE
Container Apps Integration Prep

### PR DESCRIPTION
**Pull request by MicroFish91**
Related to:  https://github.com/microsoft/vscode-azurecontainerapps/pull/170

### CODE REVIEW

**Review by bwateratmsft — DISMISSED** [review]

**Comment by MicroFish91** [comment]
There were changes made to main since the old approval, so I pushed up the new changes, will have an associated PR on container apps as well to get the new context values to match up properly

File: src/utils/azureUtils.ts [hunk 1]
```diff
@@ -18,6 +18,13 @@ import { treeUtils } from './treeUtils';
 export function createGroupConfigFromResource(resource: AppResource, subscriptionId: string | undefined): GroupingConfig {
     const id = nonNullProp(resource, 'id');
     const unknown = localize('unknown', 'Unknown');
+    let iconPath: TreeItemIconPath;
```
Comment (nturinski) [inline src/utils/azureUtils.ts:21]: Logic feels a little backwards to me here.  What about doing something like:

```
let iconPath = getIconPath(resource.azExtResourceType);
if (resource.azExtResourceType === AzExtResourceType.ContainerAppsEnvironment) {
    // comment as to why this is needed
        iconPath = getIconPath(AzExtResourceType.ContainerApps);
    }

File: src/utils/azureUtils.ts [hunk 2]
```diff
@@ -55,10 +62,17 @@ export function createAzureExtensionsGroupConfig(extensions: IAzExtMetadata[], s
     const azExtGroupConfigs: GroupNodeConfiguration[] = [];
     for (const azExt of extensions) {
         for (const azExtResourceType of azExt.resourceTypes) {
+            let iconPath: TreeItemIconPath;
```
Comment (nturinski) [inline src/utils/azureUtils.ts:65]: Logic feels a little backwards to me here.  What about doing something like:

```
let iconPath = getIconPath(resource.azExtResourceType);
if (resource.azExtResourceType === AzExtResourceType.ContainerAppsEnvironment) {
   // comment as to why this is needed
    iconPath = getIconPath(AzExtResourceType.ContainerApps);
 }

Comment (MicroFish91) [reply src/utils/azureUtils.ts:65]: Yeah, good call, thanks.  Been a bit frazzled today, not thinking super clearly.

File: src/utils/azureUtils.ts (hunk 1 above)
Comment (MicroFish91) [reply src/utils/azureUtils.ts:21]: I think it's actually swapped, I removed this one and it didn't work, but I removed the iconPath change in the other section and it worked

**Comment by MicroFish91** [comment]
Update: Changes made

**Review by alexweininger — APPROVED** [review]

**Review by nturinski — APPROVED** [review]

**Pull request merged**